dotnet/src/Experimental/Process.IntegrationTests.Shared/ProcessTests.cs

```diff
@@ -272,6 +272,93 @@ public async Task StepAndFanInProcessAsync()
+    /// └───────┘  │   └──────────────┘  │  └──────────────┘    └──────┘
+    ///            ▼                     ▼
+    ///        ┌───────┐             ┌───────┐
+    ///        │  4th  │             │  5th  │
+    ///        └───────┘             └───────┘
+    /// </summary>
+    /// <returns>A <see cref="Task"/></returns>
+    [Fact]
+    public async Task ProcessWith2NestedSubprocessSequentiallyAndMultipleOutputStepsAsync()
+    {
+        // Arrange
+        Kernel kernel = this._kernelBuilder.Build();
+        string lastStepName = "lastEmitterStep";
+        ProcessBuilder processBuilder = new(nameof(ProcessWith2NestedSubprocessSequentiallyAndMultipleOutputStepsAsync));
+
+        ProcessStepBuilder firstStep = processBuilder.AddStepFromType<EmitterStep>("firstEmitterStep");
+        ProcessBuilder secondStep = processBuilder.AddStepFromProcess(this.CreateLongSequentialProcessWithFanInAsOutputStep("subprocess1"));
+        ProcessBuilder thirdStep = processBuilder.AddStepFromProcess(this.CreateLongSequentialProcessWithFanInAsOutputStep("subprocess2"));
+        ProcessStepBuilder fourthStep = processBuilder.AddStepFromType<EmitterStep>("fourthStep");
+        ProcessStepBuilder fifthStep = processBuilder.AddStepFromType<EmitterStep>("fifthStep");
+        ProcessStepBuilder lastStep = processBuilder.AddStepFromType<FanInStep>(lastStepName);
+
+        processBuilder
+            .OnInputEvent(EmitterStep.InputEvent)
+            .SendEventTo(new ProcessFunctionTargetBuilder(firstStep, functionName: EmitterStep.InternalEventFunction));
+        firstStep
+            .OnEvent(EmitterStep.EventId)
+            .SendEventTo(secondStep.WhereInputEventIs(EmitterStep.InputEvent));
+        secondStep
+            .OnEvent(ProcessTestsEvents.OutputReadyPublic)
+            .SendEventTo(new ProcessFunctionTargetBuilder(fourthStep, functionName: EmitterStep.PublicEventFunction))
+            .SendEventTo(thirdStep.WhereInputEventIs(EmitterStep.InputEvent));
+
+        firstStep
+            .OnEvent(EmitterStep.EventId)
+            .SendEventTo(new ProcessFunctionTargetBuilder(lastStep, parameterName: "firstInput"));
+        thirdStep
+            .OnEvent(ProcessTestsEvents.OutputReadyPublic)
+            .SendEventTo(new ProcessFunctionTargetBuilder(lastStep, parameterName: "secondInput"))
+            .SendEventTo(new ProcessFunctionTargetBuilder(fifthStep, functionName: EmitterStep.PublicEventFunction));
+
+        KernelProcess process = processBuilder.Build();
+
+        // Act
+        string testInput = "SomeData";
+        var processHandle = await this._fixture.StartProcessAsync(process, kernel, new KernelProcessEvent() { Id = EmitterStep.InputEvent, Data = testInput });
+        var processInfo = await processHandle.GetStateAsync();
+
+        // Assert
+        var outputStep = (processInfo.Steps.Where(s => s.State.Name == lastStepName).FirstOrDefault() as KernelProcessStepInfo).State as KernelProcessStepState<StepState>;
```

Comment:
`var outputStep = processInfo.Steps.FirstOrDefault(s => s.State.Name == lastStepName)?.State as KernelProcessStepState<StepState>;`

Reply:
refactored and move the step state check to a common method used by all tests using this approach